code/search/src/binary_search/binary_search.js

```diff
@@ -1,18 +1,49 @@
+// implementation by looping
 function binarySearch(array, key) {
-    var lo = 0,
-        hi = array.length - 1,
-        mid,
-        element;
-    while (lo <= hi) {
-        mid = Math.floor((lo + hi) / 2, 10);
-        element = array[mid];
-        if (element < key) {
-            lo = mid + 1;
-        } else if (element > key) {
-            hi = mid - 1;
-        } else {
-            return mid;
-        }
+  var lo = 0,
+    hi = array.length - 1,
+    mid,
+    element;
+  while (lo <= hi) {
+    mid = Math.floor((lo + hi) / 2, 10);
+    element = array[mid];
+    if (element < key) {
+      lo = mid + 1;
+    } else if (element > key) {
+      hi = mid - 1;
+    } else {
+      return mid;
     }
+  }
+  return -1;
+}
+
+// implementation by recursion
+/**
+ *
+ * @param {* number[]} arr - the sorted array to be searched in
+ * @param {* number} value - the value to be searched
+ * @param {* number} low - the start index of the search range
+ * @param {* number} high - the end index of the search range
+ */
+function binarySearchByRecursion(arr, value, low, high) {
```

Comment:
In ES6, Default parameters can be placed directly in the function. ie. `...Recursion(arr, value, low = 0, high=arr.length - 1)`. Also don't call the function `binarySearchByRecursion`, just call it `binarySearchRecursion`.